src/OrchardCore/OrchardCore/Shell/Distributed/DistributedShellHostedService.cs

```diff
@@ -94,15 +94,19 @@ protected override async Task ExecuteAsync(CancellationToken stoppingToken)
                         continue;
                     }
 
-                    // If the default tenant is uninitialized, check its shared state periodically.
-                    if (defaultContext.Settings.State == TenantState.Uninitialized &&
-                        defaultTenantSyncingPeriod++ > DefaultTenantSyncingPeriod)
+                    // Manage the syncing period of the default tenant while it is 'Uninitialized'.
+                    defaultTenantSyncingPeriod = defaultContext.Settings.State == TenantState.Uninitialized
+                        ? defaultTenantSyncingPeriod++
+                        : 0;
+
+                    // Check periodically if the default tenant is no more 'Uninitialized'.
+                    if (defaultTenantSyncingPeriod++ > DefaultTenantSyncingPeriod)
```

Comment:
The naming is confusing while both are default, can we use `DefaultDefaultTenantSyncingPeriod` or `DefaultTenantSyncingPeriodDefault`

May be `DefaultSyncingPeriodForDefaultTenant`

Reply:
Yes, they should not have the same name, one is a period the other is a second counter.

Reply:
The PR is looks good except this tiny comment ;)